cmake/compiler/host-gcc/target.cmake

```diff
@@ -57,6 +57,24 @@ if (NOT CONFIG_X86_64)
   LIST(APPEND TOOLCHAIN_LIBS gcc)
 endif()
 
+set(NOSTDINC "")
+
+# Note that NOSYSDEF_CFLAG may be an empty string, and
+# set_ifndef() does not work with empty string.
+if(NOT DEFINED NOSYSDEF_CFLAG)
+  set(NOSYSDEF_CFLAG -undef)
+endif()
+
+foreach(file_name include)
```

Comment:
this code is now in 3 different place, agree this should be maintained only once.

Reply:
Regarding the location of a single location, I don't think `cmake/extensions.cmake` is the right place: It is pretty clean as-is and it would be a shame to taint it with compiler-specifics. That would counter some of the toolchain abstraction we're doing.

My proposal would be to introduce a new file `cmake/compiler/gcc-common/target.cmake` or similar. 
Probably `cmake/compiler/gcc-like/target.cmake` would be needed for stuff shared with clang.

Reply:
I'm hoping that it would be possible to merge host-gcc into gcc.

This would resolve the copy-paste issues across the board without the need to create new abstractions.

Reply:
Gradually moving code to a new `cmake/compiler/gcc-common/target.cmake` sounds like it would progressively reduce the size of `gcc/target.cmake` and `host-gcc/target.cmake` and make that merge easier and easier over time - as opposed to slowly growing the duplication and risk of divergence  like now and making that merge more and more difficult over time.
Granted, each function avoiding copy/paste requires a new name but I wouldn't always call these names "abstractions", typically not for short snippets of code.
BTW that particular foreach is also in `clang/target.cmake`, not just gcc